matpower/matpower-model/src/main/java/com/powsybl/matpower/model/MatpowerBinReader.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.matpower.model;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import us.hebi.matlab.mat.format.Mat5;
+import us.hebi.matlab.mat.types.MatFile;
+import us.hebi.matlab.mat.types.Matrix;
+import us.hebi.matlab.mat.types.Sources;
+import us.hebi.matlab.mat.types.Struct;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.powsybl.matpower.model.MatpowerReader.MATPOWER_SUPPORTED_VERSION;
+
+/**
+ * @author Christian Biasuzzi <christian.biasuzzi@techrain.eu>
+ */
+public class MatpowerBinReader {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MatpowerBinReader.class);
+    public static final String MATPOWER_STRUCT_NAME = "mpc";
+
+    public MatpowerModel read(InputStream iStream, String caseName) throws IOException {
+        MatpowerModel model = null;
+        try (MatFile mat = Mat5.newReader(Sources.wrapInputStream(iStream)).setEntryFilter(entry -> entry.getName().equals(MATPOWER_STRUCT_NAME)).readMat()) {
+            if (mat.getNumEntries() == 0) {
+                throw new IllegalStateException("not a MATPOWER stream: a structure named '" + MATPOWER_STRUCT_NAME + "' is expected.");
+            }
+            Struct mpcStruct = mat.getStruct(MATPOWER_STRUCT_NAME);
+            Set<String> mpcNames = new HashSet<>(Arrays.asList("version", "baseMVA", "bus", "gen", "branch"));
```

Comment:
This is not the proper way to initialize a set from a list of constants. Use `Sets.of`

Reply:
isn't Sets.of java9?  I changed it with a guava method, instead.